grokability/snipe-it
# Added Proper Docker Compose and .env.docker-setup files

# Description

For a while now @svpernova09 and I have been planning on rewriting the [Docker Docs](https://snipe-it.readme.io/docs/docker) so it uses Docker compose, instead of Docker run. That requires us to change some stuff in the repo. Like having a  `docker-compose.yml` and `.env.docker` file which can be used by the user.

This Pr should help with getting less support inquiries in the Discord Docker channel and remove `docker run` from production deployments.

Thankies to @svpernova09 for helping me brainstorm about possible solutions and [providing the health checks](https://github.com/svpernova09/snipe-it-kickstart/blob/0595ad89275f7ed89a8dd248c9cdb1455e887598/docker/docker-compose.yml#L35) for the `docker-compose.yml`

## Type of change


# How Has This Been Tested?

The updated article can be read in [the docs](https://snipe-it.readme.io/suggested-edits/6643d725ca9d4e003dac4ffa/preview).
But I tested my setup, by running the following commands.
- Clone Repo
- Rename `.env.docker` to `.env`
- Change port in `.env`, if needed
- Run `docker compose up -d` and wait for startup
- Add Content
- Run `docker compose down` to remove deployment, but not the volumes
- Run `docker compose up -d` and wait for startup
- Check if content is still there
- Run `docker compose down -v` delete deployment and data

# Checklist:

- [x] I have made corresponding changes to the documentation ([Not yet merged](https://snipe-it.readme.io/suggested-edits/6643d725ca9d4e003dac4ffa), waiting for this PR to be merged first)
